### PR TITLE
Add simple tests for ArrayWithOffset marshaling.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Add some basic tests for marshalling an `ArrayWithOffset` value to and from native code.

### CODE REVIEW

File: tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Runtime.InteropServices;
+using TestLibrary;
+
+unsafe class ArrayWithOffsetTest
+{
+    private const string NativeLibrary = "ArrayWithOffsetNative";
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs]: Can we use the `nameof()` trick here instead?

File: tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    [DllImport(NativeLibrary)]
+    private static extern bool Marshal_InOut(int* expected, [In, Out] ArrayWithOffset actual, int numElements, int* newValue);
+
+    [DllImport(NativeLibrary, EntryPoint = "Marshal_Invalid")]
+    private static extern bool Marshal_In(ArrayWithOffset invalidArray);
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs]: I don't really understand what this is trying to do here. Do you have any argument as to what we are validating/confirming?

File: tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    Assert.AreEqual(newValue[j], array[j]);
+                }
+            }
+
+            ArrayWithOffset arrayWithOffset = new ArrayWithOffset(new int[]{ 1 }, 0);
+
+            Assert.Throws<MarshalDirectiveException>(() => Marshal_In(arrayWithOffset));
+            Assert.Throws<MarshalDirectiveException>(() => Marshal_Invalid(ref arrayWithOffset));
+        }
+        catch (System.Exception e)
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs]: Remove `System`.

File: tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs (hunk 2 above)
Comment (jkoritzinsky) [reply tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs]: We're validating that we throw a `MarshalDirectiveException` when we don't have both `[In, Out]` and are just using the default (which I believe shows up as `[In]`.

**Pull request merged**